spec/acceptance/steps/links_indexing_steps.rb

```diff
@@ -0,0 +1,114 @@
+step 'I create a draft content item' do
+  @content_id = SecureRandom.uuid
+  @base_path = "/" + SecureRandom.hex
+
+  payload = {
+    title: 'Hallo',
+    base_path: @base_path,
+    schema_name: 'detailed_guide',
+    document_type: 'detailed_guide',
+    publishing_app: 'end-to-end-publishing-test',
+    rendering_app: 'end-to-end-publishing-test',
+    locale: 'en',
+    routes: [
+      { path: @base_path, type: 'exact' }
+    ]
+  }
+
+  Services.publishing_api.put_content(@content_id, payload)
+
+  sleep 0.1
```

Comment:
Waiters can work for testing negatives as well.
It's obviously much less pleasant, because your tests run for ages, but once you can distribute your tests across many workers using a box packing algo a sprinkling of this sort of test won't massively affect the total runtime.

I strongly discourage the adoption of natural-language tools to support acceptance testing.
There is not a requirement to use these tests as a communication tool with non-programmers, which is just about the only reason one might have for breaking the changeability and discoverability we are used to with plain ruby.

Stick with RSpec, build a dsl abstraction layer to drive the system, and write tests in terms of that custom dsl. It's much less work in the long term.

I sincerely believe that the use of cucumber-style general purpose acceptance test languages is a major reason that acctest suites become useless and unmaintained over time. I think David Heath had this experience last time we tried this on govuk.